src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2572,6 +2586,23 @@ internal void AddOrWrapTupleMembers(SourceMemberContainerTypeSymbol type)
             }
         }
 
+        private sealed class SimpleProgramEntryPointInfo
+        {
+            public readonly ImmutableArray<SynthesizedSimpleProgramEntryPointSymbol> SimpleProgramEntryPoints;
```

Comment:
Is it true that in non-error scenarios, there are either 0 or 1 elements in here?

Reply:
It feels like if we expect exactly one of these in a non-error scenario we might put a field on this class for that "single" one, and only use this field for error recovery. However, since it's not likely that very many of these will be created, it doesn't seem like an important change.

Reply:
Correct. There's one entry per file with top-level statement.
However, there's code (`SynthesizedSimpleProgramEntryPointSymbol? GetSimpleProgramEntryPoint(CSharpCompilation compilation, CompilationUnitSyntax compilationUnit, bool fallbackToMainEntryPoint)`) which needs to access all of the entry point symbols. That's why I couldn't store just zero/one here.